src/core/utils/useLoans.tsx

```diff
@@ -0,0 +1,138 @@
+
+type Loans = {
+  overdue: LoanType[];
+  soonOverdue: LoanType[];
+  farFromOverdue: LoanType[];
+  isLoading: boolean;
+  isError: boolean;
+  sortedByDate?: LoanType[];
+  stackedMaterialsDueDates?: string[];
+};
+
+type UseLoansType = {
+  all: Loans & {
+    loans: LoanType[];
+  };
+  fbs: Loans;
+  publizon: Loans;
+};
+
+type UseLoans = () => UseLoansType;
+
+const useLoans: UseLoans = () => {
+  const config = useConfig();
+  const {
+    data: loansFbs,
+    isLoading: isLoadingFbs,
+    isError: isErrorFbs
+  } = useGetLoansV2();
+  const {
+    data: loansPublizon,
+    isLoading: isLoadingPublizon,
+    isError: isErrorPublizon
+  } = useGetV1UserLoans();
+  const {
+    colorThresholds: { warning }
+  } = config<ThresholdType>("thresholdConfig", {
+    transformer: "jsonParse"
+  });
+
+  const loansIsLoading = isLoadingFbs || isLoadingPublizon;
+  const loansIsError = isErrorFbs || isErrorPublizon;
+
+  // map loans to same type
+  const mappedLoansFbs = loansFbs ? mapFBSLoanToLoanType(loansFbs) : [];
+  const mappedLoansPublizon = loansPublizon?.loans
+    ? mapPublizonLoanToLoanType(loansPublizon.loans)
+        // TODO: is it necessary to filter out loans without dueDate?
+        // there are loans without dueDate in the publizon MOCK data
+        .filter((item) => item.dueDate)
+    : [];
+
+  // Combine all loans from both FBS and Publizon
+  const loans = sortByDueDate([...mappedLoansFbs, ...mappedLoansPublizon]);
+
+  // Combine "overdue loans" from both FBS and Publizon
+  const loansOverdueFBS = filterLoansOverdue(mappedLoansFbs);
+  const LoansOverduePublizon = filterLoansOverdue(mappedLoansPublizon);
+  const loansOverdue = sortByDueDate([
+    ...loansOverdueFBS,
+    ...LoansOverduePublizon
+  ]);
+
+  // combine "soon overdue" loans from both FBS and Publizon
+  const loansSoonOverdueFBS = filterLoansSoonOverdue(mappedLoansFbs, warning);
+  const loansSoonOverduePublizon = filterLoansSoonOverdue(
+    mappedLoansPublizon,
+    warning
+  );
+  const loansSoonOverdue = sortByDueDate([
+    ...loansSoonOverdueFBS,
+    ...loansSoonOverduePublizon
+  ]);
+
+  // combine "far from overdue" loans from both FBS and Publizon
+  const loansFarFromOverdueFBS = filterLoansNotOverdue(mappedLoansFbs, warning);
+  const loansFarFromOverduePublizon = filterLoansNotOverdue(
+    mappedLoansPublizon,
+    warning
+  );
+  const loansFarFromOverdue = sortByDueDate([
+    ...loansFarFromOverdueFBS,
+    ...loansFarFromOverduePublizon
+  ]);
+
+  // The due dates are used for the stacked materials
+  // The stacked materials view shows materials stacked by
+  // due date, and for this we need a unique list of due dates
+  const loansSortedByDateFbs = sortByDueDate(mappedLoansFbs);
+  const loansSortedByDatePublizon = sortByDueDate(mappedLoansPublizon);
+
+  // list of all due dates used for the stacked materials
+  const stackedMaterialsDueDatesFbs = getDueDatesLoan(mappedLoansFbs);
+  return {
+    all: {
+      loans,
```

Comment:
Consider dropping the `loans` property, make the `sortedByDueDate` property required and then assign `loans` to `all.sortedByDueDate`.

1. As far as I can tell `loans` is already sorted on L69.
2. Required attributes should be preferred to optional
3. It removed the special case of of the `loans` property on `all`.